twisted/conch/test/test_keys.py

```diff
@@ -564,6 +564,49 @@ def test_notEqual(self):
         self.assertTrue(rsa1 != None)
 
 
+    def test_fingerprintdefault(self):
+        """
+        Test that the fingerprint method returns fingerprint in
+        C{md5-hex} format by default.
+        """
+        self.assertEqual(keys.Key(self.rsaObj).fingerprint(),
+            '3d:13:5f:cb:c9:79:8a:93:06:27:65:bc:3d:0b:8f:af')
+        self.assertEqual(keys.Key(self.dsaObj).fingerprint(),
+            '63:15:b3:0e:e6:4f:50:de:91:48:3d:01:6b:b3:13:c1')
+
+
+    def test_fingerprint_md5_hex(self):
+        """
+        fingerprint method generates key fingerprint in C{md5-hex}
+        format if explicitly specified.
+        """
+        self.assertEqual(keys.Key(self.rsaObj).fingerprint('md5-hex'),
+            '3d:13:5f:cb:c9:79:8a:93:06:27:65:bc:3d:0b:8f:af')
+        self.assertEqual(keys.Key(self.dsaObj).fingerprint('md5-hex'),
+            '63:15:b3:0e:e6:4f:50:de:91:48:3d:01:6b:b3:13:c1')
+
+
+    def test_fingerprintsha256(self):
+        """
+        fingerprint method generates key fingerprint in C{sha256-base64}
+        format if explicitly specified.
+        """
+        self.assertEqual(keys.Key(self.rsaObj).fingerprint('sha256-base64'),
+            'ryaugIFT0B8ItuszldMEU7q14rG/wj9HkRosMeBWkts=')
+        self.assertEqual(keys.Key(self.dsaObj).fingerprint('sha256-base64'),
+            'Wz5o2YbKyxOEcJn1au/UaALSVruUzfz0vaLI1xiIGyY=')
+
+
+    def test_fingerprintBadFormat(self):
+        """
+        A C{BadFingerPrintFormat} error is raised when unsupported
+        formats are requested.
+        """
+        self.assertRaisesRegexp(keys.BadFingerPrintFormat,
```

Comment:
I think that we should use assertRaisesRegex as  assertRaisesRegexp was removed in py3.... but do we need regex here?

why not do an exact match on the exception's attributes?